pkg/pillar/hypervisor/qmp.go

```diff
@@ -15,71 +18,91 @@ import (
 // this package implements subset of
 //     https://qemu.weilnetz.de/doc/qemu-qmp-ref.html
 
-const sockTimeout = 10 * time.Second
+const (
+	sockTimeout   = 10 * time.Second
+	qmpRetries    = 5
+	qmpRetrySleep = 3 * time.Second
+)
 
-func execRawCmd(socket, cmd string) ([]byte, error) {
-	var retry = 3
+// execRawCmd will retry (if doReply is set) for qmpRetries times sleeping
+// qmpRetrySleep between attempts
+func execRawCmd(socket, cmd string, doRetry bool) ([]byte, error) {
+	var retry int
 	logrus.Debugf("executing QMP command: %s", cmd)
 	var err error
 	var monitor *qmp.SocketMonitor
 
+	if doRetry {
+		retry = qmpRetries
+	}
+	// Make sure we have at least one try left for Connect below
 	for retry >= 0 {
-		if monitor, err = qmp.NewSocketMonitor("unix", socket, sockTimeout); err == nil {
-			break
+		if monitor, err = qmp.NewSocketMonitor("unix", socket, sockTimeout); err != nil {
+			retry = retry - 1
+			time.Sleep(qmpRetrySleep)
+			continue
 		}
-		retry = retry - 1
-		time.Sleep(time.Second)
+		break
 	}
-
 	if err != nil {
 		return nil, err
 	}
-
-	if err = monitor.Connect(); err != nil {
+	// Make sure we have at least one try left for Connect
+	if retry == 0 {
+		retry = 1
```

Comment:
But what's the point of sharing the retry counter between 2 different actions? Can't we just reset it here?...
And regarding the current version of the counter reset: if `retry` is 0, it is still enough to run a cycle of the following loop (it's `for retry >= 0`, not `for retry > 0`). So setting it to 1 gives 2 attempts to `.Connect`. Is it on purpose?

Reply:
@eriknordmark, I guess you could miss this comment =)

Reply:
We're sharing it to limit the total sleep time. I haven't check whether a wait forever will cause the watchdog to fire - depends which goroutine in domainmgr is calling these functions. But in general we want to limit the waits to avoid any watchdog issues.

I'll fix the 0/-1 issue - good catch. (It is correctly named retry counter - not try counter - so zero means one try. But after the first loop the counter could be negative.)

Reply:
Fixed.

Reply:
The only way to get `retry` negative after the first loop - is when we start the last iteration with `retry` equal to 0, then fail here:
https://github.com/lf-edge/eve/blob/d75c7e7077c8cde7058ad5246d582d6ba2659baf/pkg/pillar/hypervisor/qmp.go#L39-L40
go to decrease the counter, and exit the loop. But in this case, we will exit the function here:
https://github.com/lf-edge/eve/blob/d75c7e7077c8cde7058ad5246d582d6ba2659baf/pkg/pillar/hypervisor/qmp.go#L46-L48
So, `retry` is at least if we rich this point:
https://github.com/lf-edge/eve/blob/d75c7e7077c8cde7058ad5246d582d6ba2659baf/pkg/pillar/hypervisor/qmp.go#L53
So, the entire counter reset can be skipped if we want to share the counter between two different functions.

Reply:
Sorry for that endless loop discussion...

Reply:
Fixed